public/app/features/query/components/QueryGroup.tsx

```diff
@@ -220,6 +310,18 @@ export class QueryGroup extends PureComponent<Props, State> {
             </>
           )}
         </div>
+        {config.featureToggles.queryLibrary && (
+          <>
+            <div className={styles.dataSourceRow}>
+              <InlineFormLabel htmlFor="saved-query-picker" width={'auto'}>
+                Saved query
+              </InlineFormLabel>
+              <div className={styles.dataSourceRowItem}>
+                <SavedQueryPicker current={this.state.savedQueryUid} onChange={this.onChangeSavedQuery} />
+              </div>
+            </div>
+          </>
+        )}
       </div>
```

Comment:
We should carefully review these changes / verify functionality given how core this component is in Grafana